modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/GodotEnums.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+
+namespace Godot.SourceGenerators
+{
+    // TODO: May need to think about compatibility here. Could Godot change these values between minor versions?
```

Comment:
There shouldn't be changes to the existing enum values between minor versions, but there might be new enum values that would be added.

There _might_ be compat breaking changes before 4.0 is released though as especially `PropertyHint` and `PropertyUsageFlags` might benefit from a cleanup pass to remove obsolete ones / rename some / reorder in a better structure.

For `VariantType`, I think it's unlikely to change at this stage.

Would CI pick it up if any enum value is added to core and not reflected here, or if values go out of sync?

Reply:
We can generate these structs in `bindings_generator.cpp` just like we do for the API. This way, it will be up-to-date, and if there are usages of removed values in the project, the CI will fail as I added a step to build the C# projects.